Add blog post on the Go SIG's goals for 2025 (#5966)
open-telemetry/opentelemetry.io: content/en/blog/2025/go-2025-goals.md

Reviewer: Rename the file so that the slug is `2025/go-goals.md`.

Author: https://github.com/open-telemetry/opentelemetry.io/pull/5966/commits/e1876de59db1082fc19afd9c157599e2892388b3

---
title: OpenTelemetry Go 2025 Goals
linkTitle: Go Goals 2025
date: 2025-01-21
author: >-
  [Tyler Yahn](https://github.com/MrAlias) (Splunk)
sig: SIG Go
# prettier-ignore
cSpell:ignore: Yahn dashpole pellared otelhttp dmathieu otelhttp codeboten otellogr otellogrus otelslog otelzap otelzerolog
---

> As we kick off 2025, the OpenTelemetry Go team has come together to set a

Reviewer 2: ```suggestion
As we kick off 2025, the [OpenTelemetry Go](https://github.com/open-telemetry/opentelemetry-go) team has come together to set a
```

opt-out model, allowing better observability of Go runtimes.

### Logs API Stability

- Priority: Yearly goal
- Tracking Project:
  [Go: Logs (GA)](https://github.com/orgs/open-telemetry/projects/43)
- Sponsor: [@pellared](https://github.com/pellared)

Stabilizing the Logs API is crucial for providing a logging solution that aligns
with OpenTelemetry’s overarching goals. Currently, a non-stable "beta"
implementation of this API is provided in the
[`log` package](https://pkg.go.dev/go.opentelemetry.io/otel/log), along with
many bridges to popular logging packages:

- [`otellogr`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otellogr)
- [`otellogrus`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otellogrus)
- [`otelslog`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelslog)
- [`otelzap`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelzap)
- [`otelzerolog`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelzerolog)

The Go SIG plans to continue its effort in developing the upstream
specification. Work to stabilize the OpenTelemetry Go implementation depends on

> this upstream development, including the development of the event signal.

Reviewer 3: nit: I do not think that "Events" are a separate signal. I think that calling it like that can cause confusion.

```suggestion
this upstream development, including the addition of Events.
```

- Sponsors: [@MrAlias](https://github.com/MrAlias)
  [@codeboten](https://github.com/codeboten)

This effort focuses on enabling configuration of the SDK with YAML and JSON
files, making it easier for users to adopt and customize OpenTelemetry without
relying solely on environment variables or code changes. Currently, the
[`config` package](https://pkg.go.dev/go.opentelemetry.io/contrib/config)
provides and implementation of this feature. As
[file-based configuration is stabilized upstream in the specification](https://github.com/orgs/open-telemetry/projects/38),
we plan to keep `config` up-to-date with these changes and provide feedback to
its development.

## Wrapping Up

The OpenTelemetry Go team has an ambitious but focused set of goals for 2025.
These initiatives will enhance the observability landscape, improve developer
experience, and strengthen the integration of OpenTelemetry within the broader
ecosystem. We’re excited to work with the community to bring these goals to
fruition!

We want to hear from you! Let us know what is missing or what you would like to
see prioritized by commenting on

> [our tracking GitHub issue](https://github.com/open-telemetry/opentelemetry-go/issues/6175).

Reviewer 2: ```suggestion
[our tracking GitHub issue](https://github.com/open-telemetry/opentelemetry-go/issues/6175).

If you'd like to participate in any of our efforts and become a contributor to the OpenTelemetry Go SIG, join our weekly SIG meetings on Thursday alternating between 09:00 PT and 10:00 PT and our channel [#otel-go](https://cloud-native.slack.com/archives/C01NPAXACKT) on [CNCF Slack](https://slack.cncf.io).
```

---
title: OpenTelemetry Go 2025 Goals
linkTitle: Go Goals 2025

> date: 2025-01-17

Author: Proposed publish date: next Tuesday

```suggestion
date: 2025-01-21
```

author: >-
  [Tyler Yahn](https://github.com/MrAlias) (Splunk)
sig: SIG Go

> cSpell:ignore:

Author: ```suggestion
cSpell:ignore: >-
```

Try to fix spelling while not failing format check.

---

> title: OpenTelemetry Go 2025 Goals
> linkTitle: Go Goals 2025

Reviewer: Why the inconsistency in the work order. How about:

```suggestion
title: OpenTelemetry Go 2025 Goals
linkTitle: Go 2025 Goals
```

Author: https://github.com/open-telemetry/opentelemetry.io/pull/5966/commits/08720842682f744393c5b0440f50a0923c3f108c

date: 2025-01-21
author: >-
  [Tyler Yahn](https://github.com/MrAlias) (Splunk)
sig: SIG Go
# prettier-ignore
cSpell:ignore: Yahn dashpole pellared otelhttp dmathieu otelhttp codeboten otellogr otellogrus otelslog otelzap otelzerolog
---

As we kick off 2025, the
[OpenTelemetry Go](https://github.com/open-telemetry/opentelemetry-go) team has
come together to set a roadmap for the year. Our focus is on driving the
OpenTelemetry Go project forward while strengthening its integration with the

> broader OpenTelemetry ecosystem. Here's an overview of our goals, their expected
> timelines, and the key contributors supporting each initiative.
> 
> ## Goals

Reviewer: ```suggestion
broader OpenTelemetry ecosystem.

## Goals

Here's an overview of our goals, their expected timelines, and the key
contributors supporting each initiative.
```

Author: https://github.com/open-telemetry/opentelemetry.io/pull/5966/commits/c1960f8a139b6a2edfb998633ae9870497e548d8

### Logs API Stability

- Priority: Yearly goal
- Tracking Project:
  [Go: Logs (GA)](https://github.com/orgs/open-telemetry/projects/43)
- Sponsor: [@pellared](https://github.com/pellared)

Stabilizing the Logs API is crucial for providing a logging solution that aligns
with OpenTelemetry’s overarching goals. Currently, a non-stable "beta"
implementation of this API is provided in the
[`log` package](https://pkg.go.dev/go.opentelemetry.io/otel/log), along with
many bridges to popular logging packages:

- [`otellogr`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otellogr)
- [`otellogrus`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otellogrus)
- [`otelslog`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelslog)
- [`otelzap`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelzap)
- [`otelzerolog`](https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelzerolog)

The Go SIG plans to continue its effort in developing the upstream
specification. Work to stabilize the OpenTelemetry Go implementation depends on
this upstream development, including the addition of Events.

### `otelhttp` Stabilization

> Priority: Yearly goal Tracking Project:
> [Go: HTTP Semconv Migration](https://github.com/orgs/open-telemetry/projects/87)
> Sponsor: [@dmathieu](https://github.com/dmathieu)

Reviewer: ```suggestion
- Priority: Yearly goal
- Tracking Project:
  [Go: HTTP Semconv Migration](https://github.com/orgs/open-telemetry/projects/87)
- Sponsor: [@dmathieu](https://github.com/dmathieu)
```

Author: https://github.com/open-telemetry/opentelemetry.io/pull/5966/commits/65a4d58b1214241e0d39281755ea9c4ba9558f43
